src/orange/OrangeTrackView.hh

```diff
@@ -251,6 +269,51 @@ OrangeTrackView& OrangeTrackView::operator=(DetailedInitializer const& init)
     return *this;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * The current position.
+ */
+CELER_FUNCTION Real3 const& OrangeTrackView::pos() const
+{
+    return this->make_lsa().pos();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * The current direction.
+ */
+CELER_FUNCTION Real3 const& OrangeTrackView::dir() const
+{
+    return this->make_lsa().dir();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * The current volume ID (null if outside).
+ */
+CELER_FUNCTION VolumeId OrangeTrackView::volume_id() const
+{
+    return this->make_lsa().vol();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * The current surface ID.
+ */
+CELER_FUNCTION SurfaceId OrangeTrackView::surface_id() const
+{
+    return this->make_lsa().surf();
```

Comment:
The current surface is local but the next surface is local: I don't think that's right..

Reply:
I assume you mean to say "next surface is global", and that the current surface should be made global as well